WcaOnRails/app/controllers/api/v0/competitions_controller.rb

```diff
@@ -1,11 +1,85 @@
 # frozen_string_literal: true
 class Api::V0::CompetitionsController < Api::V0::ApiController
-  def show
-    competition = Competition.where(id: params[:id], showAtAll: true).first
+  def competition_from_params
+    id = params[:competition_id] || params[:id]
+    competition = Competition.find_by_id(id)
+
+    # If this competition exists, but is not publicly visible, then only show it
+    # to the user if they are able to manage the competition.
+    if competition && !competition.showAtAll && !can_manage_competition?(competition)
+      competition = nil
+    end
+
     unless competition
-      render json: { error: "Competition with id #{params[:id]} not found" }, status: 404
-      return
+      render json: { error: "Competition with id #{id} not found" }, status: 404
+      return nil
+    end
+    competition
+  end
+
+  def raise_unless_manage_competitions_scope
+    raise WcaExceptions::BadApiParameter.new("Must be logged in to specify `managed_by_me`") unless current_resource_owner
+    raise WcaExceptions::BadApiParameter.new("Missing required scope `manage_competitions`") unless current_resource_owner.doorkeeper_token.scopes.include?("manage_competitions")
+  end
+
+  def index
+    managed_by_user = nil
+    if params[:managed_by_me].present?
+      raise_unless_manage_competitions_scope
+      managed_by_user = current_resource_owner
     end
+
+    competitions = Competition.search(params[:q], params: params, managed_by_user: managed_by_user)
+    competitions = competitions.includes(:delegates, :organizers)
+
+    paginate json: competitions
+  end
+
+  def show
+    competition = competition_from_params
+    return unless competition
+
     render json: competition
   end
+
+  def can_manage_competition?(competition)
+    current_resource_owner && doorkeeper_token.scopes.exists?("manage_competitions") && current_resource_owner.can_manage_competition?(competition)
+  end
+
+  def error_if_cannot_manage_competition?(competition)
+    return true unless competition # We already rendered an error for this in competition_from_params
+
+    unless current_resource_owner
+      render json: { error: "Not authenticated" }, status: 401
+      return true
+    end
+
+    unless can_manage_competition?(competition)
+      render json: { error: "Not authorized to manage competition" }, status: 403
+      return true
+    end
+
+    false
+  end
+
+  def big_json
+    competition = competition_from_params
+    unless error_if_cannot_manage_competition?(competition)
+      render json: competition.competition_big_json.big_json
```

Comment:
1. No, I suspect you're thinking of our various join tables, which AFAIK are the "rails way" to name join tables. See http://guides.rubyonrails.org/association_basics.html#creating-join-tables-for-has-and-belongs-to-many-associations. That said, this isn't a join table, so I agree that prefixing its name with `competition_` is a bad idea.
2. The only name we have for this is ["WCA Competition JSON Format"](https://github.com/thewca/worldcubeassociation.org/wiki/WCA-Competition-JSON-Format). I'd LOVE to have a name for this. I like CCIF. Does CCIF stand for cube competition interchange format?